templates/Dockerfile.template

```diff
@@ -0,0 +1,242 @@
+  ; \
+  \
+# Check that Erlang/OTP crypto & ssl were compiled against OpenSSL correctly
+  erl -noshell -eval 'io:format("~p~n~n~p~n~n", [crypto:supports(), ssl:versions()]), init:stop().'
+
+ENV RABBITMQ_DATA_DIR=/var/lib/rabbitmq
+# Create rabbitmq system user & group, fix permissions & allow root user to connect to the RabbitMQ Erlang VM
+RUN set -eux; \
+  groupadd --gid 999 --system rabbitmq; \
+  useradd --uid 999 --system --home-dir "$RABBITMQ_DATA_DIR" --gid rabbitmq rabbitmq; \
+  mkdir -p "$RABBITMQ_DATA_DIR" /etc/rabbitmq /tmp/rabbitmq-ssl /var/log/rabbitmq; \
+  chown -fR rabbitmq:rabbitmq "$RABBITMQ_DATA_DIR" /etc/rabbitmq /tmp/rabbitmq-ssl /var/log/rabbitmq; \
+  chmod 777 "$RABBITMQ_DATA_DIR" /etc/rabbitmq /tmp/rabbitmq-ssl /var/log/rabbitmq; \
+  ln -sf "$RABBITMQ_DATA_DIR/.erlang.cookie" /root/.erlang.cookie
+
+
+ENV RABBITMQ_VERSION {{ $rabbitmq.Version }}
+ENV RABBITMQ_SHA512="{{ $rabbitmq.Sha512 }}"
+ENV RABBITMQ_HOME=/opt/rabbitmq
+
+# Add RabbitMQ to PATH, send all logs to TTY
+ENV PATH=$RABBITMQ_HOME/sbin:$PATH \
+  RABBITMQ_LOGS=-
+
+# Install RabbitMQ
+RUN set -eux; \
+  \
+  savedAptMark="$(apt-mark showmanual)"; \
+  apt-get update; \
+  apt-get install --yes --no-install-recommends \
+    ca-certificates \
+    dirmngr \
+    gnupg2 \
+    wget \
+    xz-utils \
+  ; \
+  rm -rf /var/lib/apt/lists/*; \
+  \
+  RABBITMQ_SOURCE_URL="https://github.com/rabbitmq/rabbitmq-server/releases/download/v$RABBITMQ_VERSION/rabbitmq-server-generic-unix-latest-toolchain-$RABBITMQ_VERSION.tar.xz"; \
+  RABBITMQ_PATH="/usr/local/src/rabbitmq-$RABBITMQ_VERSION"; \
+  \
+  wget --progress dot:giga --output-document "$RABBITMQ_PATH.tar.xz" "$RABBITMQ_SOURCE_URL"; \
+  \
+  mkdir -p "$RABBITMQ_HOME"; \
+  echo "$RABBITMQ_SHA512 *$RABBITMQ_PATH.tar.xz" | sha512sum --check --strict -; \
+  tar --extract --file "$RABBITMQ_PATH.tar.xz" --directory "$RABBITMQ_HOME" --strip-components 1; \
+  rm -rf "$RABBITMQ_PATH"*; \
+# Do not default SYS_PREFIX to RABBITMQ_HOME, leave it empty
+  grep -qE '^SYS_PREFIX=\$\{RABBITMQ_HOME\}$' "$RABBITMQ_HOME/sbin/rabbitmq-defaults"; \
+  sed -i 's/^SYS_PREFIX=.*$/SYS_PREFIX=/' "$RABBITMQ_HOME/sbin/rabbitmq-defaults"; \
+  grep -qE '^SYS_PREFIX=$' "$RABBITMQ_HOME/sbin/rabbitmq-defaults"; \
+  chown -R rabbitmq:rabbitmq "$RABBITMQ_HOME"; \
+  \
+  # reset apt-mark's "manual" list so that "purge --auto-remove" will remove all build dependencies
+  apt-mark auto '.*' > /dev/null; \
+  [ -z "$savedAptMark" ] || apt-mark manual $savedAptMark; \
+  find /usr/local -type f -executable -exec ldd '{}' ';' \
+    | awk '/=>/ { print $(NF-1) }' \
+    | sort -u \
+    | xargs -r dpkg-query --search \
+    | cut -d: -f1 \
+    | sort -u \
+    | xargs -r apt-mark manual \
+  ; \
+  apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \
+# verify assumption of no stale cookies
+  [ ! -e "$RABBITMQ_DATA_DIR/.erlang.cookie" ]
+
+RUN apt update && apt install -y openssl
```

Comment:
Why installed second time?

Reply:
Rabbitmq needs the module to work properly and it's removed by the script which removes all the build deps (it's also a requirement for building erlang). 

Found that would be simpler to remove all the deps and reinstalling only openssl instead of doing a long list to apt purge.